spyder/widgets/projects/tests/test_projectdialog.py

```diff
@@ -28,5 +29,29 @@ def test_project_dialog(qtbot):
     assert dlg
 
 
+@patch('spyder.widgets.projects.projectdialog.getexistingdirectory')
+def test_projectdialog_location(mock_getexistingdirectory):
+    """Run getexistingdirectory with mock values"""
+    dlg = ProjectDialog(None)
+    mock_getexistingdirectory.return_value = r"c:\a/b\\c/d"
```

Comment:
Here, you should actually use Mock/MagicMock. See for example [here](https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/variableexplorer/tests/test_collectioneditor.py).